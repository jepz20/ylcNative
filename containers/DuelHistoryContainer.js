import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import { PlayerHistory } from '../components'

class DuelHistoryContainer extends Component {
  renderPlayersHistory () {
    const { players, logs, currentDuel, onPress } = this.props

    return players.valueSeq().map(player => {
      return (
        <PlayerHistory
          key={player.id}
          logs={logs}
          playerId={player.id}
          currentDuel={currentDuel}
          onPress={onPress}
        />
      )
    })
  }

  render () {
    return (
      <View style={[styles.container, this.props.style]}>
        {this.renderPlayersHistory()}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row'
  }
})
export { DuelHistoryContainer }
