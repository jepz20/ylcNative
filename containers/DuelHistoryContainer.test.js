import React from 'react'
import { Map, Record } from 'immutable'
import renderer from 'react-test-renderer'

import { DuelHistoryContainer } from './DuelHistoryContainer'

describe('DuelHistoryContainer', () => {
  const playerRE = Record({
    id: '1',
    name: 'test 1',
    currentPoints: '8000'
  })

  test('renders with no props', () => {
    const duelHistoryContainer = renderer
      .create(<DuelHistoryContainer />)
      .toJSON()
    expect(duelHistoryContainer).toMatchSnapshot()
  })

  test('renders correctly with one player', () => {
    const players = Map({
      '1': playerRE()
    })
    const duelHistoryContainer = renderer
      .create(<DuelHistoryContainer players={players} />)
      .toJSON()
    expect(duelHistoryContainer).toMatchSnapshot()
  })

  test('renders correctly with more than 1 player', () => {
    const players = Map({
      '1': playerRE(),
      '2': playerRE({ id: 2, name: 'test 2' })
    })
    const duelHistoryContainer = renderer
      .create(<DuelHistoryContainer players={players} />)
      .toJSON()
    expect(duelHistoryContainer).toMatchSnapshot()
  })
})
