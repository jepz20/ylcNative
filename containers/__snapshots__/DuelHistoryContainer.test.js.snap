// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DuelHistoryContainer renders correctly with more than 1 player 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 5,
        "flexDirection": "row",
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={undefined}
        testID={undefined}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={undefined}
        testID={undefined}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`DuelHistoryContainer renders correctly with one player 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 5,
        "flexDirection": "row",
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={undefined}
        testID={undefined}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "grey",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
